project-koku/koku
# [COST-2001] add missing indexes to aws/azure daily summary tables for group by options

## Jira Ticket

[COST-2001](https://issues.redhat.com/browse/COST-2001)

## Description

This change will add missing indexes to `reporting_azurecostentrylineitem_daily_summary` and `reporting_awscostentrylineitem_daily_summary` based on our group by options. There is also a small change to the `migrate_schemas.py` file to account for org prefixed schemas.

## Testing

1. Checkout Branch
2. Restart Koku
3. Watch migrations run
4. Check the indexes for the table to confirm the new indexes are present, for example:
```
SELECT tablename, indexname, indexdef FROM pg_indexes WHERE schemaname = 'org1234567' AND tablename = 'reporting_azurecostentrylineitem_daily_summary_2022_09';
```
5. If you want to verify the org prefixed schema change, go back a migration: 
```
python koku/manage.py migrate_schemas reporting 0257
```
7. Then remove the org prefix change in `migrate_schemas.py`
8. reapply the migration:
```
python koku/manage.py migrate_schemas reporting
```
9. You will notice the org schema was not migrated
10. Re-do step 5 to get migrations back to a previous migration
11. add the org prefix change back and re-apply the migrations and watch it work on the org schema

## Notes

...
